python/scripts/rfi_receiver/rfi_receiver.py

```diff
@@ -822,18 +855,21 @@ def rfi_zeroing():
         time.sleep(1)
     while True:
         # Wait until the correct UTC time of the solar transit at DRAO (deals with daylight savings time)
-        t_now = datetime.datetime.utcnow()
-        t_transit = ephemeris.solar_transit(t_now)[0]
-        t_diff = datetime.datetime.utcfromtimestamp(t_transit) - t_now
+        today = datetime.datetime.utcnow().date()
+        t_day_start = datetime.datetime(today.year, today.month, today.day)
+        t_transit = ephemeris.solar_transit(t_day_start)[0]
+        t_diff = (
+            datetime.datetime.utcfromtimestamp(t_transit) - datetime.datetime.utcnow()
+        )
 
         time_to_transit_s = abs(t_diff.total_seconds())
 
         # Check if we are in the transit window, if so set downtime accordingly
         if time_to_transit_s < half_downtime_s:
+            toggle_rfi_zeroing(url, headers, False)
             downtime_s = time_to_transit_s + half_downtime_s
```

Comment:
I think better logic in here is something along the lines of this:
```python
time_now = ephemeris.ensure_unix(datetime.utcnow())

# Get the *next* transit in the future
time_to_next_transit = ephemeris.solar_transit(time_now) - time_now

# Get the *nearest* transit which we need to determine if we are still in the window
time_to_nearest_transit = ephemeris.solar_transit(time_now - 12 * 3600) - time_now + 12 * 3600

# Check if we are within the current transit window and wait until the end of it
if abs(time_to_nearest_transit) < half_window_s:
    set_rfi_zeroing(False)
    downtime_s = half_window + time_to_nearest_transit
# Otherwise, we wait until the start of the next transit window
else:
    set_rfi_zeroing(True)
    downtime_s = time_to_next_transit - half_window_s
```

Also note that call the window width `half_downtime_s` is confusing because `downtime_s` used elsewhere seems to have a different meaning, i.e. the first is the length of half of the window, but the latter is the time time to wait until the next transition. The name implies the former is just half the latter, which it isn't.